draft-ietf-quic-transport.md

```diff
@@ -703,6 +702,23 @@ use it for all subsequent Handshake ({{packet-handshake}}) and 1-RTT
 Server's Version Negotiation ({{packet-version}}) and Retry ({{packet-retry}})
 packets MUST use connection ID selected by the client.
 
+A connection ID is truncated when included in short headers.  Each endpoint
+expresses how many octets of the connection ID they require in packets that are
+sent to that endpoint.  The truncate_connection_id transport parameter
+({{transport-parameter-definitions}}) of a receiving endpoint determines the
+number of octets that are included by the sending peer.  A truncated connection
+ID includes the indicated number of octets, starting from the first octet.  An
+endpoint can request that the connection ID to be omitted entirely by setting
+truncate_connection_id to 0, or include any number of octets up to the full
+connection ID length of 17 octets.  Connection IDs are not truncated for any
+other use.
+
+Note:
+
+: The odd length of the connection ID is chosen to enable efficient construction
+  of the value using 128-bit ciphers with an additional octet for overheads such
+  as key identifiers.
+
```

Comment:
OK. Now, in the direction from server to client, are we sending the server chosen CID, or the client chosen one? Because, the client has no way to know how many bytes of the server connection ID it might need before seeing the CID. Yet the client's transport parameters are sent in the client initial packet, before seeing the value chosen by the server. So the clients only practical choices are 0 and 17...

Reply:
Yes, there is general recognition that a client can't use stateless routing.  For a client, you are right.  This is why I would prefer some symmetric in connection ID selection, but that means making a hard call with respect to the stateless reset.

Reply:
In regards to the stateless reset issue for asymmetric connection IDs, one use case is picking a connection ID so the packet is received by the optimal receive queue and doesn't have to hop threads/CPUs.  For use cases like this, it's an optimization, but the machine should be able to route the packet correctly if it chooses to.

Also, I'll note that right now, the client doesn't get to pick any portion of the connection ID, so allowing the client to pick knowing that stateless reset might not route optimally seems acceptable, since the client can always pick the same value as the server, like today.